### PR TITLE
Copy services across accounts

### PR DESCRIPTION
**Pull request by macejmic**
tested with Victor :+1: 

closes https://github.com/3scale/3scale_toolbox/issues/4

### CODE REVIEW

File: exe/3scale-copy [hunk 1]
```diff
@@ -40,40 +40,59 @@ def compare_hashes(first, second, keys)
   keys.map{ |key| first.fetch(key) } == keys.map{ |key| second.fetch(key) }
 end
 
+def fetch_option(options, key)
+  options.fetch(key) { raise OptionParser::MissingArgument, key }
+end
+
+def provider_key_from_url(url)
+  url[/\w*@{1}/][0..-2]
```
Comment (kidd) [inline exe/3scale-copy]: what's the difference between '@{1}' and '@' ?

Comment (macejmic) [inline exe/3scale-copy]: good point. no difference.

File: exe/3scale-copy [hunk 2]
```diff
@@ -40,40 +40,59 @@ def compare_hashes(first, second, keys)
   keys.map{ |key| first.fetch(key) } == keys.map{ |key| second.fetch(key) }
 end
 
+def fetch_option(options, key)
+  options.fetch(key) { raise OptionParser::MissingArgument, key }
+end
+
+def provider_key_from_url(url)
+  url[/\w*@/][0..-2]
```
Comment (mikz) [inline exe/3scale-copy]: I'd much rather see `URI(url).userinfo` than custom regex magic :)

File: exe/3scale-copy [hunk 3]
```diff
@@ -40,40 +40,59 @@ def compare_hashes(first, second, keys)
...
+end
+
+def endpoint_from_url(url)
+  url.sub /\w*@/, ''
```
Comment (mikz) [inline exe/3scale-copy]: the same here

```ruby
uri = URI(url)
uri.userinfo = nil
uri.to_s
```

File: exe/3scale-copy (hunk 2 above)
Comment (mikz) [inline exe/3scale-copy]: on second thought... we probably should allow ThreeScale::API to accept whole uri containing the provider key

File: exe/3scale-copy [hunk 4]
```diff
@@ -40,40 +40,59 @@ def compare_hashes(first, second, keys)
...
     service_id = ARGV.shift or raise OptionParser::MissingArgument, 'service_id'
 
-    service = client.show_service(service_id)
-
-    name = "#{service['name']} (copy)"
+    source_client = ThreeScale::API.new(
+      endpoint:     endpoint_from_url(source), 
+      provider_key: provider_key_from_url(source)
+    )
+    client        = ThreeScale::API.new(
```
Comment (mikz) [inline exe/3scale-copy]: created an issue for this https://github.com/3scale/3scale-api-ruby/issues/9

**Comment by mikz** [comment]
Good job!

File: exe/3scale-copy [hunk 5]
```diff
@@ -9,12 +9,12 @@ options = {}
 parser = OptionParser.new do |parser|
   parser.banner = '3scale copy <command> [options]'
 
-  parser.on('-e', '--endpoint ENDPOINT', "Endpoint") do |domain|
-    options[:endpoint] = domain
+  parser.on('-source', '--source SOURCE', "Source") do |domain|
```
Comment (mikz) [inline exe/3scale-copy]: why changing it from `-e` to `-source` ? I'd understand `--endpoint` to `--source`. but why the one dash shortcut to have full word? or is optparser smart and shows just first letter?

File: exe/3scale-copy (hunk 2 above)
Comment (macejmic) [inline exe/3scale-copy]: didn't know about that ```URI(url).userinfo``` :+1:

File: exe/3scale-copy (hunk 5 above)
Comment (macejmic) [inline exe/3scale-copy]: everything make sense now. will fix that later. :hamster:

Comment (mikz) [inline exe/3scale-copy]: no worries, will fix it in master and do a release

**Pull request merged**